controllers/dspipeline_controller.go

```diff
@@ -162,7 +164,7 @@ func (r *DSPAReconciler) DeleteResourceIfItExists(ctx context.Context, obj clien
 //+kubebuilder:rbac:groups=argoproj.io,resources=workflowtaskresults,verbs=create;patch
 //+kubebuilder:rbac:groups=argoproj.io,resources=workflowartifactgctasks;workflowartifactgctasks/finalizers,verbs=*
 //+kubebuilder:rbac:groups=core,resources=pods;pods/exec;pods/log;services,verbs=*
-//+kubebuilder:rbac:groups=core;apps;extensions,resources=deployments;replicasets,verbs=*
+//+kubebuilder:rbac:groups=core;apps;extensions,resources=deployments;deployments/finalizers;replicasets,verbs=*
```

Comment:
As discussed in Slack, I think this may not be needed.

Reply:
@mprahl This permission was added to prevent the following deployment error, which occurs when the controller attempts to set blockOwnerDeletion on a resource without having the required permission to set finalizers on its owner:
`ERROR Reconciler error {"controller": "datasciencepipelinesapplication", "controllerGroup": "datasciencepipelinesapplications.opendatahub.io", "controllerKind": "DataSciencePipelinesApplication", "DataSciencePipelinesApplication": {"name":"sample","namespace":"dspa"}, "namespace": "dspa", "name": "sample", "reconcileID": "567e6648-9be8-497c-974b-a37f89eb747e", "error": "deployments.apps \"ds-pipelines-webhook\" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on: , <nil>"}`